### ISSUE TITLE
"Go to definition" causes error in Spyder 4.1.5 on Windows

### ISSUE THREAD

**Issue by BenZickel** [spyder-ide/spyder#13807]
### What steps will reproduce the problem?

Any use of "Go to definition" in Windows with Spyder 4.1.5 raises this error.

### Traceback
```python-traceback
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\editor\widgets\codeeditor.py", line 1440, in handle_go_to_definition
    def_range = position['range']
TypeError: list indices must be integers or slices, not str

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\ProgramData\Anaconda3\lib\logging\__init__.py", line 992, in emit
    msg = self.format(record)
  File "c:\ProgramData\Anaconda3\lib\logging\__init__.py", line 838, in format
    return fmt.format(record)
  File "c:\ProgramData\Anaconda3\lib\logging\__init__.py", line 575, in format
    record.message = record.getMessage()
  File "c:\ProgramData\Anaconda3\lib\logging\__init__.py", line 338, in getMessage
    msg = msg % self.args
ValueError: incomplete format
Call stack:
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\app\start.py", line 223, in <module>
    main()
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\app\start.py", line 219, in main
    mainwindow.main()
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\app\mainwindow.py", line 3651, in main
    mainwindow = run_spyder(app, options, args)
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\app\mainwindow.py", line 3567, in run_spyder
    app.exec_()
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\completion\languageserver\client.py", line 497, in on_msg_received
    self.req_reply[req_id](None, {'params': []})
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\completion\languageserver\plugin.py", line 549, in receive_response
    self.COMPLETION_CLIENT_NAME, req_id, response)
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\completion\plugin.py", line 123, in receive_response
    self.howto_send_to_codeeditor(req_id)
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\completion\plugin.py", line 171, in howto_send_to_codeeditor
    self.skip_and_send_to_codeeditor(req_id)
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\completion\plugin.py", line 205, in skip_and_send_to_codeeditor
    self.gather_and_send_to_codeeditor(request_responses)
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\completion\plugin.py", line 273, in gather_and_send_to_codeeditor
    response_instance.handle_response(req_type, responses)
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\editor\widgets\codeeditor.py", line 1019, in handle_response
    handler(params)
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\editor\widgets\codeeditor.py", line 1457, in handle_go_to_definition
    "Error when processing go to definition")
  File "c:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\editor\widgets\codeeditor.py", line 1050, in log_lsp_handle_errors
    logger.error('%', 1, stack_info=True)
Message: '%'
Arguments: (1,)
```

## Versions

* Spyder version: 4.1.5 
* Python version: 3.6.4
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  0.5.2 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.16.1 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  0.7.0 (OK)
paramiko >=2.4.0               :  2.4.2 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.4 (OK)
psutil >=5.3                   :  5.7.2 (OK)
pygments >=2.0                 :  2.2.0 (OK)
pylint >=1.0                   :  1.8.2 (OK)
pyls >=0.34.0;<1.0.0           :  0.34.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.2 (OK)
qtconsole >=4.6.0              :  4.7.6 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.2.1 (OK)
spyder_kernels >=1.9.4;<1.10.0 :  1.9.4 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.1 (OK)

# Optional:
cython >=0.21                  :  0.27.3 (OK)
matplotlib >=2.0.0             :  2.2.2 (OK)
numpy >=1.7                    :  1.18.1 (OK)
pandas >=0.13.1                :  0.24.2 (OK)
scipy >=0.17.0                 :  1.4.1 (OK)
sympy >=0.7.3                  :  1.1.1 (OK)
```

**Comment by ccordoba12** [comment]
Hey @BenZickel, thanks for reporting. Could you provide a source code example that it's generating this error?

Otherwise it's won't be possible for us to come up with a solution for it.

**Comment by BenZickel** [comment]
In order to reproduce this just open a new file in the editor and put the below line in it:

from collections import namedtuple

Now try to select namedtuple and use "Go to definition" on it.

**Comment by ccordoba12** [comment]
> Python version: 3.6.4

Maybe the issue is with Python 3.6.

@dalthviz, could you try to reproduce this on Windows?

**Comment by dalthviz** [comment]
@ccordoba12 testing with Python 3.6.4 seems like `Go to definition` isn't working (although no error is raised for me). Just in case this is the env used for the testing:

<details>

```
# packages in environment at C:\Users\Daniel\Anaconda3\envs\py36:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                   py36_0
argh                      0.26.2                   py36_0
asn1crypto                1.3.0                    py36_0
astroid                   2.3.3                    py36_0
atomicwrites              1.3.0                    py36_1
attrs                     19.3.0                     py_0
autopep8                  1.4.4                      py_0
babel                     2.8.0                      py_0
backcall                  0.1.0                    py36_0
bcrypt                    3.1.7            py36he774522_0
bleach                    3.1.0                    py36_0
ca-certificates           2020.7.22                     0
certifi                   2020.6.20                py36_0
cffi                      1.13.2           py36h7a1dbc1_0
chardet                   3.0.4                 py36_1003
cloudpickle               1.2.2                      py_0
colorama                  0.4.3                      py_0
cryptography              2.8              py36h7a1dbc1_0
decorator                 4.4.1                      py_0
defusedxml                0.6.0                      py_0
diff-match-patch          20181111                   py_0
docutils                  0.15.2                   py36_0
entrypoints               0.3                      py36_0
flake8                    3.8.3                      py_0
future                    0.18.2                   py36_0
icu                       58.2                 ha66f8fd_1
idna                      2.8                      py36_0
imagesize                 1.2.0                      py_0
importlib-metadata        1.7.0                    py36_0
importlib_metadata        1.3.0                    py36_0
intervaltree              3.0.2                      py_0
ipykernel                 5.1.3            py36h39e3cac_1
ipython                   7.11.1           py36h39e3cac_0
ipython_genutils          0.2.0                    py36_0
isort                     4.3.21                   py36_0
jedi                      0.17.1                   py36_0
jinja2                    2.10.3                     py_0
jpeg                      9b                   hb83a4c4_2
jsonschema                3.2.0                    py36_0
jupyter_client            5.3.4                    py36_0
jupyter_core              4.6.1                    py36_0
keyring                   20.0.0                   py36_0
lazy-object-proxy         1.4.3            py36he774522_0
libpng                    1.6.37               h2a8f88b_0
libsodium                 1.0.16               h9d3ae62_0
libspatialindex           1.9.3                h33f27b4_0
markupsafe                1.1.1            py36he774522_0
mccabe                    0.6.1                    py36_1
mistune                   0.8.4            py36he774522_0
more-itertools            8.0.2                      py_0
nbconvert                 5.6.1                    py36_0
nbformat                  4.4.0                    py36_0
numpydoc                  0.9.2                      py_0
openssl                   1.1.1g               he774522_1
packaging                 20.0                       py_0
pandoc                    2.2.3.2                       0
pandocfilters             1.4.2                    py36_1
paramiko                  2.6.0                    py36_0
parso                     0.7.0                      py_0
pathtools                 0.1.2                      py_1
pexpect                   4.7.0                    py36_0
pickleshare               0.7.5                    py36_0
pip                       19.3.1                   py36_0
pluggy                    0.13.1                   py36_0
prompt_toolkit            3.0.2                      py_0
psutil                    5.6.7            py36he774522_0
pycodestyle               2.6.0                      py_0
pycparser                 2.19                     py36_0
pydocstyle                4.0.1                      py_0
pyflakes                  2.2.0                      py_0
pygments                  2.5.2                      py_0
pylint                    2.4.4                    py36_0
pynacl                    1.3.0            py36h62dcd97_0
pyopenssl                 19.1.0                   py36_0
pyparsing                 2.4.6                      py_0
pyqt                      5.9.2            py36h6538335_2
pyrsistent                0.15.6           py36he774522_0
pysocks                   1.7.1                    py36_0
python                    3.6.4                h0c2934d_3
python-dateutil           2.8.1                      py_0
python-jsonrpc-server     0.3.2                      py_0
python-language-server    0.34.1                   py36_0
pytz                      2019.3                     py_0
pywin32                   227              py36he774522_1
pywin32-ctypes            0.2.0                    py36_0
pyyaml                    5.2              py36he774522_0
pyzmq                     18.1.0           py36ha925a31_0
qdarkstyle                2.8.1                      py_0
qt                        5.9.7            vc14h73c81de_0
qtawesome                 0.6.0                      py_0
qtconsole                 4.6.0                      py_1
qtpy                      1.9.0                      py_0
requests                  2.22.0                   py36_1
rope                      0.14.0                     py_0
rtree                     0.8.3                    py36_0
setuptools                44.0.0                   py36_0
sip                       4.19.8           py36h6538335_0
six                       1.13.0                   py36_0
snowballstemmer           2.0.0                      py_0
sortedcontainers          2.1.0                    py36_0
sphinx                    2.3.1                      py_0
sphinxcontrib-applehelp   1.0.1                      py_0
sphinxcontrib-devhelp     1.0.1                      py_0
sphinxcontrib-htmlhelp    1.0.2                      py_0
sphinxcontrib-jsmath      1.0.1                      py_0
sphinxcontrib-qthelp      1.0.2                      py_0
sphinxcontrib-serializinghtml 1.1.3                      py_0
spyder                    4.1.5                    py36_0
spyder-kernels            1.9.4                    py36_0
sqlite                    3.30.1               he774522_0
testpath                  0.4.4                      py_0
tornado                   6.0.3            py36he774522_0
traitlets                 4.3.3                    py36_0
typed-ast                 1.4.0            py36he774522_0
ujson                     1.35             py36hfa6e2cd_0
urllib3                   1.25.7                   py36_0
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.16.27012          hf0eaf9b_1
watchdog                  0.9.0                    py36_1
wcwidth                   0.1.7                    py36_0
webencodings              0.5.1                    py36_1
wheel                     0.33.6                   py36_0
win_inet_pton             1.1.0                    py36_0
wincertstore              0.2              py36h7fe50ca_0
wrapt                     1.11.2           py36he774522_0
yaml                      0.1.7                hc54c509_2
yapf                      0.28.0                     py_0
zeromq                    4.3.1                h33f27b4_3
zipp                      0.6.0                      py_0
zlib                      1.2.11               h62dcd97_3


```

</details>

**Comment by ccordoba12** [comment]
Could you update python-jsonrpc-server to 0.3.4 and try again?

I don't think that will make much of a difference, but just in case.

**Comment by dalthviz** [comment]
Checking that didn't work, however if I change my interpreter to one I have with python 3.7.9 the go to definition works as expected (event if Spyder was started from the Python 3.6.4 env).

Something else is that the hover tooltips doesn't work either . Checking a little bit more, seems like the problem is the PyLS and Jedi versions with Python 3.6. If I change Jedi to 0.15.2 and python-language-server to 0.31.10 the hover and Go to definition work as expected while using Python 3.6.

**Comment by ccordoba12** [comment]
Ok, then this is a regression introduced by Jedi. @dalthviz, could you come up with a reproducible example and report it to Jedi? Thanks!

**Comment by dalthviz** [comment]
Checking a little bit more seems like an update to jedi 0.17.2 fixes the issues (seems like jedi 0.17.1 has an issue when running under Python 3.6):

```python
import jedi
source = '''
import json
json.lo'''
script = jedi.Script(source, path='example.py')
completions = script.complete(3, len('json.lo'))
Traceback (most recent call last):

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\inference\compiled\subprocess\__init__.py", line 261, in _send
    is_exception, traceback, result = pickle_load(self._get_process().stdout)

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\_compatibility.py", line 396, in pickle_load
    return pickle.load(file, encoding='bytes')

EOFError: Ran out of input


During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\api\environment.py", line 75, in _get_subprocess
    info = self._subprocess._send(None, _get_info)

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\inference\compiled\subprocess\__init__.py", line 273, in _send
    stderr,

InternalError: The subprocess C:\Users\Daniel\Anaconda3\envs\py36\python.exe has crashed (EOFError('Ran out of input',), stderr=).


During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "<ipython-input-1-ef826d01cd16>", line 5, in <module>
    script = jedi.Script(source, path='example.py')

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\api\__init__.py", line 185, in __init__
    project, environment=environment, script_path=self.path

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\inference\__init__.py", line 90, in __init__
    self.compiled_subprocess = environment.get_inference_state_subprocess(self)

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\api\environment.py", line 114, in get_inference_state_subprocess
    return InferenceStateSubprocess(inference_state, self._get_subprocess())

  File "C:\Users\Daniel\Anaconda3\envs\py36\lib\site-packages\jedi\api\environment.py", line 80, in _get_subprocess
    exc))

InvalidPythonEnvironment: Could not get version information for 'C:\\Users\\Daniel\\Anaconda3\\envs\\py36\\python.exe': InternalError("The subprocess C:\\Users\\Daniel\\Anaconda3\\envs\\py36\\python.exe has crashed (EOFError('Ran out of input',), stderr=).",)
```

I think the issue is related with davidhalter/jedi#1627

With Jedi 0.17.2everything (as far as I tested) works as expected. Maybe we need to change the Jedi dependency version from =0.17.1 to = 0.17.2 ?
